framework/Optimizers/Optimizer.py

```diff
@@ -92,44 +55,84 @@ def __init__(self):
-    self.inputInfo['SampledVars'     ]  = self.values               # this is the location where to get the values of the sampled variables
+    self.inputInfo['SampledVars']       = self.values               # this is the location where to get the values of the sampled variables
     self.constants                      = {}                        # dictionary of constants variables
-    self.FIXME                          = False                     # FIXME flag
     self.printTag                       = self.type                 # prefix for all prints (optimizer type)
-
-    self._endJobRunnable                = sys.maxsize               # max number of inputs creatable by the optimizer right after a job ends
-
+    self.submissionQueue                = {}                        # by traj, a place (deque) to store points that should be submitted some time after they are discovered
+    #functions and dataojbects
     self.constraintFunction             = None                      # External constraint function, could be not present
+    self.solutionExport                 = None                      #This is the data used to export the solution (it could also not be present)
     self.mdlEvalHist                    = None                      # Containing information of all model evaluation
     self.objSearchingROM                = None                      # ROM used internally for fast loss function evaluation
-
+    #multilevel
+    self.multilevel                     = False                     # indicates if operating in multilevel mode
+    self.mlBatches                      = {}                        # dict of {batchName:[list,of,vars]} that defines input subspaces
+    self.mlSequence                     = []                        # list of batch names that determines the order of convergence.  Last entry is converged most often and fastest (innermost loop).
+    self.mlDepth                        = {}                        # {traj: #} index of current recursion depth within self.mlSequence, must be initialized to None
+    self.mlStaticValues                 = {}                        # by traj, dictionary of static values for variables in fullOptVars but not in optVars due to multilevel
+    self.mlActiveSpaceSteps             = {}                        # by traj, integer to track iterations performed in optimizing the current, active subspace
+    self.mlBatchInfo                    = {}                        # by batch, by traj, info includes 'lastStepSize','gradientHistory','recommendToGain'
+    #stateful tracking
     self.nextActionNeeded               = (None,None)               # tool for localStillReady to inform localGenerateInput on the next action needed
-
+    self.status                         = {}                        # by trajectory, ("string-based status", arbitrary, other, entries)
```

Comment:
tl dr; Using a dictionary of strings is sufficiently efficient, though it still causes me to pause on this code. Since this code is under active development and it does not incur a performance penalty, it should work for now, but should be revisited for maintainability later. It would be very easy for these strings and the conditionals where they are used to get out of sync.